src/UrlHelper.php

```diff
@@ -63,34 +78,55 @@ public function __invoke($routeName = null, array $params = [])
             $basePath = '';
         }
 
+        // Get the options to be passed to the router
+        $routerOptions = array_key_exists('router', $options) ? $options['router'] : [];
+
         if ($routeName === null) {
-            return $basePath . $this->generateUriFromResult($params, $result);
+            return $basePath . $this->generateUriFromResult($routeParams, $result, $routerOptions);
         }
 
-        if ($result) {
-            $params = $this->mergeParams($routeName, $result, $params);
+        $reuseResultParams = !isset($options['reuse_result_params']) || (bool) $options['reuse_result_params'];
```

Comment:
Pretty sure that once you get composer able to install dependencies, this line will be flagged as a CS issue (`!` needs to be followed by a space). Try aliasing the zend-expressive-router dep so you can run the CS checks, just to eliminate another potential hurdle later.